core/src/main/java/org/opensearch/sql/expression/function/OpenSearchFunctions.java

```diff
@@ -27,16 +23,6 @@
 
 @UtilityClass
 public class OpenSearchFunctions {
```

Comment:
in case of refactoring relevance queries. could you think how to let the OpenSearch engine register the functions also? The reasons are.
1. Core engine should not aware of OpenSearch engine at all.
2. Currently, the relevance functions throw exception if it is not OpenSearch engine. ""OpenSearch defined function [%s] is only supported in WHERE and HAVING clause." during query execution stage, the exception should been throw at analysis stage instead.

Reply:
@penghuo can we split moving OpenSearch-specifics out of `:core` into a separate issue? There is another PR blocked on this one.

1. I did start moving `OpenSearchFunctions` from `:core` to `:opensearch`. It requires updating Spring usage to initialize `ExpressionConfig.functionRepository` with configurations from two different modules. This directly affects PPL and SQL endpoints. It also results in ExpressionAnalyzer test being split across two modules.

2. The exception is related to current code structure  as well. It is thrown out of `valueOf` to raise error for invalid queries like `SELECT MATCH_PHRASE(...` We need to introduce separate where-clause expression and select-statement expression in the grammar to make this exception unnecessary. Currently there's only one `expression` rule used in both cases.

I'd like to do these as separate PRs.

Reply:
make sense. Thanks.